osquery/tables/system/darwin/system_info.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <string>
+
+#include <osquery/sql.h>
+#include <osquery/tables.h>
+
+#include <osquery/logger.h>
+#include <osquery/database.h>
+#include "osquery/database/query.h"
+#include "osquery/sql/sqlite_util.h"
+#include "osquery/core/conversions.h"
+
+#include <CoreFoundation/CoreFoundation.h>
+#include <IOKit/IOKitLib.h>
+
+namespace osquery {
+namespace tables {
+
+
+std::string getHostUUID() {
+  std::string uuid;
+  auto status = getDatabaseValue(kPersistentSettings, "hostIdentifier", uuid);
+  if (!status.ok()) {
+    // Unable to retrieve a uuid from the database.
+    // Note that this will always occur with osqueryi.
+    return "<unknown>";
+  }
+  else if (uuid.size() == 0) {
+    // There was no uuid stored in the database, generate one and store it.
+    uuid = osquery::generateHostUuid();
+    auto status = setDatabaseValue(kPersistentSettings, "hostIdentifier", uuid);
+    if (!status.ok()) {
+      // Unable to store our new uuid, so ignore that we created one
+      return "<unknown>";
+    } else {
+      // Log that we created and successfully stored a new uuid
+      VLOG(1) << "Using uuid " << uuid << " as host identifier";
+    }
+  }
+  return uuid;
+}
+
+std::string getCpuId() {
+  io_service_t service = IOServiceGetMatchingService(
+    kIOMasterPortDefault,
+    IOServiceMatching("IOPlatformExpertDevice"));
+  CFStringRef serialNumber = (CFStringRef)IORegistryEntryCreateCFProperty(
+    service,
+    CFSTR("IOPlatformSerialNumber"),
+    kCFAllocatorDefault,
+    0);
+
+  auto cpu_id = stringFromCFString(serialNumber);
+  if (cpu_id.compare("") == 0) {
```

Comment:
Use: `cpu_id.size() == 0` or better yet: `cpu_id.empty()` for constant time empty checking!